build.gradle

```diff
@@ -0,0 +1,179 @@
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:0.9.+'
+        classpath files('lib/gradle-witness.jar')
+    }
+}
+
+configurations {
+    provided
+    compile
+}
+
+apply plugin: 'witness'
+apply plugin: 'java'
+
+repositories {
+    jcenter()
+    mavenLocal() // TODO: use lib/ instead?
+    ivy {
+        url 'https://downloads.freenetproject.org/'
+        layout 'pattern', {
+            artifact '/latest/[module][revision].[ext]'
+        }
+    }
+}
+
+sourceSets {
+    main.compileClasspath += configurations.provided
+    test.compileClasspath += configurations.provided
+}
+
+sourceSets {
+    main {
+        java {
+            srcDir 'src/'
+        }
+    }
+    test {
+        java {
+            srcDir 'test/'
+        }
+    }
+}
+
+compileJava {
+    sourceCompatibility = 1.7
+    targetCompatibility = 1.7
+    archivesBaseName = "freenet"
+}
+
+def gitrev
+task buildInfo {
+    try {
+        def cmd = "git describe --always --abbrev=4 --dirty"
+        def proc = cmd.execute()
+        gitrev = proc.text.trim()
+    } catch (java.io.IOException e) {
+        gitrev = "@unknown@"
+    }
+}
+
+// This isn't used anywhere yet
+// I am not sure of whether we should parse and use the info here or generate the file!
+def classpathDependencies = []
+task parseManifest {
+    Properties properties = new Properties()
+    new File('dependencies.properties').withInputStream {
+        properties.load(it)
+    }
+    for (String s in properties.stringPropertyNames().findAll {
+    it.endsWith(".type") && properties.getProperty(it) == "CLASSPATH" }) {
+        def dep = s.split("\\.")[0]
+        def dep_sha256 = properties.getProperty("${dep}.sha256")
+        def dep_version = properties.getProperty("${dep}.version")
+        def dep_filename = properties.getProperty("${dep}.filename")
+	classpathDependencies.add([ dep, dep_filename, dep_version, dep_sha256 ])
+    }
+}
+
+jar {
+    manifest {
+        attributes("Permissions": "all-permissions")
+        attributes("Application-Name": "Freenet REference Daemon")
+        attributes("Required-Ext-Version": 29)
+        attributes("Recommended-Ext-Version": 29)
+        attributes([
+                       "Specification-Title": "Freenet",
+                       "Specification-Version": "0.7.5",
+                       "Specification-Vendor": "freenetproject.org",
+                       "Implementation-Title": "Freenet",
+                       "Implementation-Version": "0.7.5 ${gitrev}",
+                       "Implementation-Vendor": "freenetproject.org",
+                   ], "common")
+    }
+}
+jar.dependsOn buildInfo
+
+task copyResourcesToClasses2 {
```

Comment:
yeah, but that means changing the ant scripts... which I am not keen on doing atm. I'd like us to have both build systems for a few releases